plaso/parsers/shared/unified_logging/tracev3.py

```diff
@@ -0,0 +1,1221 @@
+            chars = raw_data[:int(flags_width_precision[1:])]
+          chars = binascii.hexlify(chars, ' ').decode('utf-8').upper()
+        else:
+          logger.error(
+            'Unknown data specifier: {}'.format(custom_specifier))
+          output += 'Invalid specifier'
+          i += 1
+          continue
+        output += chars
+      elif specifier == 'p':
+        if data_size == 0:
+          if data_type & constants.FIREHOSE_ITEM_STRING_PRIVATE:
+            output += '<private>'
+          else:
+            logger.error(
+                'Size 0 in pointer fmt {0:s} // data {1!s}'.format(
+                    format_string, data_item))
+            output += 'Invalid specifier'
+            i += 1
+            continue
+        else:
+          if data_size == 2:
+            data_map = uint16_data_type_map
+          elif data_size == 4:
+            data_map = uint32_data_type_map
+          elif data_size == 8:
+            data_map = uint64_data_type_map
+          else:
+            logger.error(
+                'Unknown data_size for pointer: {0:d} // fmt {1:s}'.format(
+                    data_size, format_string))
+            output += 'Invalid specifier'
+            i += 1
+            continue
+          try:
+            number = self._ReadStructureFromByteStream(raw_data, 0, data_map)
+          except ValueError:
+            pass
+          if flags_width_precision:
+            logger.error('Width/Precision not supported for *p specifiers')
+      else:
+        output += 'Unknown Specifier'
+
+      i += 1
+
+    if last_start_index < len(format_string):
+      output += format_string[last_start_index:].replace('%%', '%')
+
+    return output
+
+  def _ReadHeader(self, file_object, file_offset):
+    """Reads a Header.
+
+    Args:
+      file_object (file): file-like object.
+      file_offset (int): offset of the catalog data relative to the start of the
+        file.
+
+    Raises:
+      ParseError: if the chunk header cannot be read.
+    """
+    data_type_map = self._GetDataTypeMap('tracev3_header')
+
+    self.header, _ = self._ReadStructureFromFileObject(file_object, file_offset,
+                                                       data_type_map)
+
+    logger.debug('Header data: CT {0:d} // Bias {1:d}'.format(
+        self.header.continuous_time, self.header.bias_in_minutes))
+    logger.debug('SI Info: BVS: {0:s} // HMS: {1:s}'.format(
+        self.header.systeminfo_subchunk.systeminfo_subchunk_data
+        .build_version_string, self.header.systeminfo_subchunk
+        .systeminfo_subchunk_data.hardware_model_string))
+    logger.debug('Boot UUID: {0:s}'.format(
+        self.header.generation_subchunk.generation_subchunk_data.boot_uuid.hex))
+    logger.debug('TZ Info: {0:s}'.format(
+        self.header.timezone_subchunk.timezone_subchunk_data.path_to_tzfile))
+
+    self.boot_uuid_ts = aul_time.GetBootUuidTimeSync(
+        self.timesync_parser.records,
```

Comment:
Why keep all the timesync records in memory while you only seem to be needing/using the ones corresponding to the boot identifier (UUID)?